pkg/cli/flags.go

```diff
@@ -784,20 +784,15 @@ func init() {
 	}
 
 	// Commands that print tables.
-	tableOutputCommands := append(
-		[]*cobra.Command{
-			sqlShellCmd,
-			genSettingsListCmd,
-			genMetricListCmd,
-			demoCmd,
-			statementBundleRecreateCmd,
-			debugListFilesCmd,
-			debugJobTraceFromClusterCmd,
-			debugZipCmd,
-		},
-		demoCmd.Commands()...)
-	tableOutputCommands = append(tableOutputCommands, nodeCmds...)
-	tableOutputCommands = append(tableOutputCommands, authCmds...)
+	tableOutputCommands := []*cobra.Command{
+		genSettingsListCmd,
+		genMetricListCmd,
+		debugListFilesCmd,
+		debugZipCmd,
+		decommissionNodeCmd,
+		recommissionNodeCmd,
+		drainNodeCmd,
+	}
```

Comment:
Removing these commands from `tableOutputCommands` as they will be handled when we iterate through `sqlCmds` on line 761.

Reply:
i see -- let's not add format to `AddSQLFlags` then. ideally, we don't want to add this option to subcommands not expecting it (i.e. the sqlCmds group). i think the safer fix is:

how about we try adding
```
// --format
cliflagcfg.VarFlagDepth(1, sqlCmd.PersistentFlags(), &cfg.ExecCtx.TableDisplayFormat, cliflags.TableDisplayFormat)
```

to here instead https://github.com/cockroachdb/cockroach/blob/b1bf99cf6cd6f4ba8e5ed0f87009f0c746b6a8c1/pkg/cmd/cockroach-sql/main.go#L90

Reply:
Good point. I've amended the commit with the new solution. I've tested it locally and it's working as expected. Let me know what you think.